pkg/dao/environments.go

```diff
@@ -136,11 +136,38 @@ func (r environmentDaoImpl) Search(orgID string, request api.ContentUnitSearchRe
 	// Set to default request limit if null or request limit max (500) if greater than max
 	request = checkRequestLimit(request)
 
-	// FIXME 103 Once the URL stored in the database does not allow
-	//           "/" tail characters, this could be removed
-	urls := handleTailChars(request)
 	uuids := request.UUIDs
 
+	// check that repository uuids exist
+	for _, uuid := range uuids {
+		found := models.RepositoryConfiguration{}
+		if err := r.db.
+			Where("uuid = ?", uuid).
+			First(&found).
+			Error; err != nil {
+			return []api.SearchEnvironmentResponse{}, &ce.DaoError{
+				BadValidation: true,
+				Message:       "Could not find repository with UUID: " + uuid,
+			}
+		}
+	}
+	// check that repository urls exist and handle tail chars
+	urls := make([]string, len(request.URLs))
+	for _, url := range request.URLs {
+		url = models.CleanupURL(url)
+		urls = append(urls, url)
+		found := models.Repository{}
+		if err := r.db.
+			Where("url = ?", url).
+			First(&found).
+			Error; err != nil {
+			return []api.SearchEnvironmentResponse{}, &ce.DaoError{
+				BadValidation: true,
+				Message:       "Could not find repository with URL: " + url,
+			}
+		}
+	}
```

Comment:
What do you think about turning these functions into shared functions between rpms, package groups, and environments?

Reply:
i think that is a great idea :D will fix

Reply:
added those helpers, is that what you had in mind?

Reply:
yep!